### PR TITLE
fix(webpack): paths plugin modification to not allow multiple requests more than once

### PR DESCRIPTION
**Pull request by jkuri**
closes https://github.com/angular/angular-cli/issues/3441

### CODE REVIEW

**Comment by hansl** [comment]
Can you add an e2e test for this PR?

**Comment by jkuri** [comment]
@hansl working on it.

**Comment by hansl** [comment]
What's the status for those tests failing?

**Comment by jkuri** [comment]
solution itself is not okay. I am suspicious of a bug in webpack resolver, but not sure. I'll try to find a solution for this elsewhere `path-plugin.ts` file since previous regex was just fine.

**Comment by chrisfitz** [comment]
FWIW, the solution I've been using to get the paths to work in my non-angular-cli webpack configuration prior to attempting to integrate `@ngtools/webpack` as the typescript loader was to use `ts.nodeModuleNameResolver` against the request and use the result of that (if any) as the new request. So far, it's worked with all the scenarios I've tried it with (including the ones I mentioned in #3441), I'm just not entirely sure if most optimal way of doing so.

**Comment by jkuri** [comment]
@chrisfitz thanks for the tip! @hansl can you please check again?

File: packages/@ngtools/webpack/src/paths-plugin.ts [hunk 1]
```diff
@@ -95,8 +95,12 @@ export class PathsPlugin implements Tapable {
         if (onlyModule) {
           aliasPattern = new RegExp(`^${excapedAlias}$`);
         } else {
-          let withStarCapturing = excapedAlias.replace('\\*', '(.*)');
-          aliasPattern = new RegExp(`^${withStarCapturing}`);
+          if (alias !== '*') {
+            let withStarCapturing = excapedAlias.replace('\\*', '(.*)');
+            aliasPattern = new RegExp(`^${withStarCapturing}$`);
+          } else {
+            aliasPattern = new RegExp(`^${excapedAlias}$`);
```
Comment (hansl) [inline packages/@ngtools/webpack/src/paths-plugin.ts]: So the pattern now is `^\\*$` which only matches a star? That seems really wrong.

Comment (jkuri) [inline packages/@ngtools/webpack/src/paths-plugin.ts]: reverted.

**Pull request closed without merging**